AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java

```diff
@@ -1546,8 +1546,11 @@ public int compare(JSONObject a, JSONObject b) {
 
         try{
             for (JSONObject n : models) {
-                long modID = n.getLong("id");
-                cardCount.add(col.getModels().nids(col.getModels().get(modID)).size());
+                if (isCancelled()) {
+                    Timber.e("doInBackgroundLoadModels :: Cancelled");
+                    return new TaskData(false);
```

Comment:
@Arthur-Milchior I think this just needs a comment here and it can merge 
```suggestion
                    // onPostExecute should not be called if canceled, return should not matter. Using false.
                    return new TaskData(false);
```

Reply:
Comment added. Returning null since we do not actually care about the value